pgrx-sql-entity-graph/src/used_type.rs

```diff
@@ -608,6 +613,143 @@ fn resolve_option_inner(
+
+                            let ty = type_for_args(without_type_args, comp_ty, err_ty);
+                            Ok((ty, sql))
+                        },
+                        // Option<default!(composite_type!(..))> isn't valid. If the user wanted the default to be `NULL` they just don't need a default.
+                        "default" => Err(syn::Error::new(mac.span(), "`Result<default!(T, \"my_default\"), E>` not supported, choose `Result<T, E>` for a default of `NULL`, or `default!(T, default)` for a non-NULL default")),
+                        _ => Ok((syn::Type::Path(original), None)),
+                    }
+                }
+                syn::Type::Path(arg_type_path) => {
+                    let last = arg_type_path.path.segments.last().ok_or(syn::Error::new(
+                        arg_type_path.span(),
+                        "No last segment in type path",
+                    ))?;
+                    match last.ident.to_string().as_str() {
+                        // Result<Option<composite_type!(..)>>
+                        // Result<Option<Vec<composite_type!(..)>>>>
+                        "Option" => {
+                            let (inner_ty, expr) = resolve_option_inner(arg_type_path)?;
+                            let wrapped_ty = type_for_args(without_type_args, inner_ty, err_ty);
+                            Ok((wrapped_ty, expr))
+                        }
+                        // Result<Vec<composite_type!(..)>>
+                        // Result<Vec<Option<composite_type!(..)>>>
+                        "Vec" => {
+                            let (inner_ty, expr) = resolve_vec_inner(arg_type_path)?;
+                            let wrapped_ty = type_for_args(without_type_args, inner_ty, err_ty);
+                            Ok((wrapped_ty, expr))
+                        }
+                        // Result<VariadicArray<composite_type!(..)>>
+                        // Result<VariadicArray<Option<composite_type!(..)>>>
+                        "VariadicArray" => {
+                            let (inner_ty, expr) = resolve_variadic_array_inner(arg_type_path)?;
+                            let wrapped_ty = type_for_args(without_type_args, inner_ty, err_ty);
+                            Ok((wrapped_ty, expr))
+                        }
+                        // Result<Array<composite_type!(..)>>
+                        // Result<Array<Option<composite_type!(..)>>>
+                        "Array" => {
+                            let (inner_ty, expr) = resolve_array_inner(arg_type_path)?;
+                            let wrapped_ty = type_for_args(without_type_args, inner_ty, err_ty);
+                            Ok((wrapped_ty, expr))
+                        }
+                        // Result<..>
+                        // TODO - evaluate, should this error rather than proceeding?
+                        _ => Ok((syn::Type::Path(original), None)),
```

Comment:
Hmm, what's the `..` a stand-in for? This is... probably fine? If we're in doubt, however, we should be returning an error.

Reply:
After digging deeper, this should be the case selected for any `Result<SomePlainOldData>` or `Result<SomePlainOldData, E>`, i.e. the `T` in `Result<T,E>` is not a container type.